server/webhook_worker.go

```diff
@@ -80,3 +83,44 @@ func (ww webhookWorker) process(msg *webhookMessage) (err error) {
 
 	return nil
 }
+
+func (wh *webhook) FetchReporterNotification(bb []byte, reporter *jira.User) error {
+	if wh.eventTypes.ContainsAny("event_created_comment") || wh.eventTypes.ContainsAny("event_updated_comment") {
+		jwhook := &JiraWebhook{}
+		err := json.Unmarshal(bb, &jwhook)
+		if err != nil {
+			return err
+		}
+		if jwhook.Issue.ID == "" {
+			return ErrWebhookIgnored
+		}
+
+		commentAuthor := mdUser(&jwhook.Comment.UpdateAuthor)
+
+		whook := &webhook{
+			JiraWebhook: jwhook,
+			eventTypes:  NewStringSet(eventCreatedComment),
+			headline:    fmt.Sprintf("%s **commented** on %s", commentAuthor, jwhook.mdKeySummaryLink()),
+			text:        truncate(quoteIssueComment(jwhook.Comment.Body), 3000),
+		}
+		jwh := whook.JiraWebhook
+
+		commentAuthor = mdUser(&jwh.Comment.UpdateAuthor)
+
+		commentMessage := fmt.Sprintf("%s **commented** on %s:\n>%s", commentAuthor, jwh.mdKeySummaryLink(), jwh.Comment.Body)
+		if reporter == nil ||
+			(reporter.Name != "" && reporter.Name == jwh.User.Name) ||
+			(reporter.AccountID != "" && reporter.AccountID == jwh.Comment.UpdateAuthor.AccountID) {
+			return nil
+		}
+		wh.notifications = append(wh.notifications, webhookUserNotification{
```

Comment:
We should make sure no this user is not already in the `wh.notifications` array. If they are in the array, we should just return instead of adding them here

Reply:
@mickmister 
i see the webhook will be called individually so there won't be duplicates correct me if i am wrong

Reply:
I would look through the code and find all the ways a notification can be added to this array. I believe the current state would be:

- The user was mentioned in the comment
- The user is the assignee
- The user is the reporter

So basically if they were mentioned or are the assignee, they will already be in the array here.

Reply:
@mickmister seems a silly question.
 when you say to check if the notification array and assure the notification is already there 
 
 ```
 webhookUserNotification{
		jiraUsername:  reporter.Name,
		jiraAccountID: reporter.AccountID,
		message:       commentMessage,
		postType:      PostTypeComment,
		commentSelf:   jwhook.Comment.Self,
		recipientType: recipientTypeReporter,
	}
 ```
 this is the notification we append , so should i check the array with the username ,accountID , postType and recipientType ?

Reply:
We just need to check username and accountID, since those are the identifiers for the user receiving the notification. The goal here is that one user should receive only one notification for this webhook request.

Reply:
Sure